File: ios/Vxin/Data/Models/CallLog.swift
import Foundation

/// 通话记录（GET /api/users/me/call-logs）。对齐 web CallHistory / 后端 getCallLogs。
struct CallLog: Decodable, Identifiable {
    let id: String
    var type: String = "audio"        // audio | video
    var status: String = "completed"  // completed | missed | canceled | rejected | ongoing
    var direction: String = "out"     // out | in
    var duration: Int = 0             // 秒
    var startedAt: Double = 0
    var endedAt: Double = 0
    var createdAt: Double = 0
    var peerId: String = ""
    var peerName: String = ""
    var peerAvatar: String = ""

    enum CodingKeys: String, CodingKey {
        case id, type, status, direction, duration
        case startedAt = "started_at"
        case endedAt = "ended_at"
        case createdAt = "created_at"
        case peerId = "peer_id"
        case peerName = "peer_name"
        case peerAvatar = "peer_avatar"
    }
    init(from decoder: Decoder) throws {
        let c = try decoder.container(keyedBy: CodingKeys.self)
        id = try c.decode(String.self, forKey: .id)
        type = (try? c.decode(String.self, forKey: .type)) ?? "audio"
        status = (try? c.decode(String.self, forKey: .status)) ?? "completed"
        direction = (try? c.decode(String.self, forKey: .direction)) ?? "out"
        duration = (try? c.decode(Int.self, forKey: .duration)) ?? 0
        startedAt = (try? c.decode(Double.self, forKey: .startedAt)) ?? 0
        endedAt = (try? c.decode(Double.self, forKey: .endedAt)) ?? 0
        createdAt = (try? c.decode(Double.self, forKey: .createdAt)) ?? 0
        peerId = (try? c.decode(String.self, forKey: .peerId)) ?? ""
        peerName = (try? c.decode(String.self, forKey: .peerName)) ?? ""
        peerAvatar = (try? c.decode(String.self, forKey: .peerAvatar)) ?? ""
    }
}

File: android/app/src/main/java/com/vxin/app/data/model/CallLog.kt
package com.vxin.app.data.model

import kotlinx.serialization.Serializable

/**
 * 通话记录（GET /api/users/me/call-logs）。对齐 web CallHistory / 后端 getCallLogs。
 * direction: out=拨出 in=来电；status: completed/missed/canceled/rejected/ongoing。
 */
@Serializable
data class CallLog(
    val id: String,
    val type: String = "audio",       // audio | video
    val status: String = "completed", // completed | missed | canceled | rejected | ongoing
    val direction: String = "out",    // out | in
    val duration: Int = 0,            // 秒
    val started_at: Long = 0,
    val ended_at: Long = 0,
    val created_at: Long = 0,
    val peer_id: String = "",
    val peer_name: String = "",
    val peer_avatar: String = "",
)
